source/common/http/http1/codec_impl.cc

```diff
@@ -1252,16 +1209,17 @@ Status ServerConnectionImpl::checkHeaderNameForUnderscores() {
 ClientConnectionImpl::ClientConnectionImpl(Network::Connection& connection, CodecStats& stats,
                                            ConnectionCallbacks&, const Http1Settings& settings,
                                            const uint32_t max_response_headers_count)
-    : ConnectionImpl(connection, stats, settings, HTTP_RESPONSE, MAX_RESPONSE_HEADERS_KB,
+    : ConnectionImpl(connection, stats, settings, MessageType::Response, MAX_RESPONSE_HEADERS_KB,
                      max_response_headers_count, formatter(settings)) {}
 
 bool ClientConnectionImpl::cannotHaveBody() {
   if (pending_response_.has_value() && pending_response_.value().encoder_.headRequest()) {
     ASSERT(!pending_response_done_);
     return true;
-  } else if (parser_.status_code == 204 || parser_.status_code == 304 ||
-             (parser_.status_code >= 200 && parser_.content_length == 0 &&
-              !(parser_.flags & F_CHUNKED))) {
+  } else if (parser_->statusCode() == 204 || parser_->statusCode() == 304 ||
```

Comment:
optional, do we want these returning the envoy status code values as long as we have an API?
I'm pro minimal churn but I'm also pro good APIs so I leave it up to you :-P

Reply:
Added a follow-up TODO to add in next PR